File: c++/Sorting/src/bubblesort.cpp
#include "bubblesort.hpp"
#include <iostream>
#include <string>

BubbleSort::BubbleSort(){

}

BubbleSort::~BubbleSort(){

}

int* BubbleSort::sort(int array[], int size){

    for(unsigned int i = 0; i < size; i++) {

        for (unsigned int j = i+1; j < size; j++)
        {
            if(array[i] > array[j]) {
                int temp = array[j];
                array[j] = array[i];
                array[i] = temp;
            }
        }
    }
    return array;
}

void BubbleSort::printArray(int array[], int size) {
    std::string str_arr;
    for (unsigned int i = 0; i < size; i++)
    {
        str_arr.append(" ").append(std::to_string(array[i]));
    }
     std::cout << "{" << str_arr << "}" << std::endl;
}

File: python/Algorithms/Sorting/bubblesort.py
class BubbleSort:

    def __init__(self):
        pass

    def sort(self, arr):
        for i in range(0, len(arr)-1):
            for j in range(i+1, len(arr)-1):
                if(arr[i] > arr[j]):
                    temp = arr[i]
                    arr[i] = arr[j]
                    arr[j] = temp
